### DISCUSSION TITLE
How  use http request to start/resume a workflow

### DISCUSSION THREAD

**Discussion by hyzx86** [OrchardCMS/OrchardCore#9782] [Q&A] [answered]
I have created a workflow definition like this:
How can I send the `correlationId` when start request ?

Because my workflow requires multiple instances at the same time, and multiple levels of approval are required in the process
So I want to isolate them using Correlation Id

![image](https://user-images.githubusercontent.com/15613121/123396966-7bb25580-d5d4-11eb-89dc-1fabd2eaad30.png)

**Comment by weirdyang** [comment, marked as answer]
You need to use the `correlate task` to set a value that will be used as the correlation id. an alternative will be to use the `signal event` and generate `signal url`

**Reply by hyzx86** [reply]
So, if I want to use SPA access to process workflow, do I need to return a signal URL to the SPA? And then , I have to store the `signal url` and wait next approver process the application

**Reply by sfmskywalker** [reply]
Exactly.

**Comment by hyzx86** [comment]
Hi @sfmskywalker,
Remember that you can specify the `CorrelationId` from the Http Header before in OC . Has it been removed in the new version? Or is it only available in elsa-workflow, because I remembered it wrong?

**Reply by sfmskywalker** [reply]
I had a brief look at the `HttpWorkflowController` in the Workflows module, and it appears that correlation is inly supported when triggering workflows using the `SignalEvent` activity, and not using the `HttpRequestEvent` activity.

It appears though that it wouldn't be too hard to add:

1. Add a `CorrelationId` and/or `WorkflowInstanceId` property to the `WorkflowPayload` class.
2. Update the `GenerateUrl` action in `HttpWorkflowController` to accept an optional correlation ID and/or workflow instance ID and pass that into the `WorkflowPayload` construction.
3. Update the `Invoke` action in `HttpWorkflowController` starting from line 164 (the branch that handles resuming HTTP workflows) to select only those workflows instances that match the incoming correlation ID or workflow instance ID (when provided).

**Reply by sfmskywalker** [reply]
Alternatively, the suggestion from @weirdyang will achieve the same result.

**Reply by hyzx86** [reply]
Got it , Thanks for your replay.
